R/SDistribution_Arrdist.R

```diff
@@ -310,30 +310,28 @@ Arrdist <- R6Class("Arrdist",
     .pdf = function(x, log = FALSE) {
       "pdf, data, wc" %=% gprm(self, c("pdf", "x", "which.curve"))
       mat <- .extCurve(pdf, wc)
-      out <- t(C_Vec_WeightedDiscretePdf(
-        x, matrix(data, ncol(mat), private$.ndists), t(mat)))
+      out <- t(C_Vec_WeightedDiscretePdf(x, data, t(mat)))
```

Comment:
@RaphaelS1 so I think I understand what the `C_Vec_WeightedDiscretePdf` does => gets a `pdf` matrix and subsets it to the `x` times (points) instead of the `data` it has. It's just the names of `x` and `data` are not so informative in this context. Some points for discussion:
1) Should we consider some name changing?
2) These matrix tranposes maybe slow down things? Wouldn't be more sense if the code worked as `C_Vec_WeightedDiscretePdf(x, data, mat))`?

Reply:
Yes maybe but not in this PR

Reply:
I think (1) would make sense to do now, (2) might be trickier

Reply:
for (1) => https://github.com/xoopR/distr6/issues/297